from launch import LaunchDescription
from launch_ros.actions import Node


def generate_launch_description():
    return LaunchDescription(
        [
            Node(package="webots_spot", executable="detect_dex_board", output="screen"),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0",
                    "0",
                    "0",
                    "0",
                    "-0.707",
                    "0",
                    "0.707",
                    "dex_board_green",
                    "linear_front",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0.05",
                    "0.3",
                    "0",
                    "0",
                    "-0.707",
                    "0",
                    "0.707",
                    "dex_board_green",
                    "front_left",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0.05",
                    "-0.3",
                    "0",
                    "0",
                    "-0.707",
                    "0",
                    "0.707",
                    "dex_board_green",
                    "front_right",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0.0",
                    "0.15",
                    "0",
                    "-0.270600",
                    "-0.653300",
                    "-0.270600",
                    "0.653300",
                    "dex_board_green",
                    "angled_left",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0.0",
                    "-0.15",
                    "0",
                    "0.270600",
                    "-0.653300",
                    "0.270600",
                    "0.653300",
                    "dex_board_green",
                    "angled_right",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0",
                    "0",
                    "0",
                    "0",
                    "-0.707",
                    "0",
                    "0.707",
                    "dex_board_blue",
                    "omni_front",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0.15",
                    "0.15",
                    "-0.05",
                    "-0.3535534",
                    "0.3535534",
                    "-0.1464466",
                    "0.8535534",
                    "omni_front",
                    "top_left",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "0.15",
                    "-0.15",
                    "-0.05",
                    "0.3535534",
                    "0.3535534",
                    "0.1464466",
                    "0.8535534",
                    "omni_front",
                    "top_right",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "-0.15",
                    "0.15",
                    "-0.05",
                    "-0.3535534",
                    "-0.3535534",
                    "0.1464466",
                    "0.8535534",
                    "omni_front",
                    "bottom_left",
                ],
            ),
            Node(
                package="tf2_ros",
                executable="static_transform_publisher",
                output="screen",
                arguments=[
                    "-0.15",
                    "-0.15",
                    "-0.05",
                    "0.3535534",
                    "-0.3535534",
                    "-0.1464466",
                    "0.8535534",
                    "omni_front",
                    "bottom_right",
                ],
            ),
        ]
    )
